railgun/internal/manager/controllerdef.go

```diff
@@ -199,49 +199,60 @@ func setupControllers(logger logr.Logger, mgr manager.Manager, proxy proxy.Proxy
 		Disabling Ingress controller`)
 	}
 
-	kongClusterPluginGVR := schema.GroupVersionResource{
-		Group:    konghqcomv1.SchemeGroupVersion.Group,
-		Version:  konghqcomv1.SchemeGroupVersion.Version,
-		Resource: "kongclusterplugins",
-	}
-	if ctrlutils.CRDExists(mgr.GetClient(), kongClusterPluginGVR) {
-		logger.Info("kongclusterplugins.configuration.konghq.com v1beta1 CRD available on cluster.")
-		controller := ControllerDef{
-			IsEnabled: &c.KongClusterPluginEnabled,
-			Controller: &configuration.KongV1KongClusterPluginReconciler{
-				Client:           mgr.GetClient(),
-				Log:              ctrl.Log.WithName("controllers").WithName("KongClusterPlugin"),
-				Scheme:           mgr.GetScheme(),
-				Proxy:            proxy,
-				IngressClassName: c.IngressClassName,
-			},
+	if c.KongClusterPluginEnabled == util.EnablementStatusEnabled {
```

Comment:
From a UX perspective this change makes sense to me because it avoids us from logging _anything_ about cluster plugins if the user didn't enable them, which I think is good. :+1: 

My only concern here is that I'm not sure what we want to do with "AutoEnablement". Auto is not disabled, so we obviously can't treat them the same, so we at least have to do _something_ right?

```suggestion
	if c.KongClusterPluginEnabled == util.EnablementStatusEnabled || c.KongClusterPluginEnabled == util.EnablementStatusAuto {
```

This I think highlights for me the problem with the current enablement status options (over just using boolean enabled/disabled), but at least in this case I think it will make this work as intended.

Reply:
This suggestion still breaks the user priority coz of the "||" syntax, also, the c.KongClusterPluginEnabled value assignment should have been handled under the config/FlagSetting module.

I mean, default controller configure each controller as "util.EnablementStatusAuto", if user specify (enable/disable), we should overwrite c.KongClusterPluginEnabled = util.EnablementStatusEnabled.

Logic comes to here, we already know enabled or not.